### DISCUSSION TITLE
Nuxt 3 Pocketbase Auth not working

### DISCUSSION THREAD

**Discussion by RA9** [pocketbase/pocketbase#926] [Q&A]
Hey guys, I have been on this for this trying to use the plugin method from pocketbase js/SDK for integrating pocketbase into nuxt 3. Everything is working fine; except each time I tried logging it doesn't add the cookies "pb_auth" in the cookies area and for some reason, it is showing in the local storage area but the token and model are always empty.

If someone has done this before, can you please share some lights as to how it was achievable?

**Comment by ganigeorgiev** [comment]
@RA9 This is similar to https://github.com/pocketbase/js-sdk/issues/29#issuecomment-1229959528.

Modern JavaScript frameworks blurs the line between browser and server code, but **you'll need first to define how you want to interact with the PocketBase API**:

1. Only in a server-side nuxt context
2. Only client-side in the browser context
3. Both

The Nuxt example from https://github.com/pocketbase/js-sdk#ssr-integration shows a sample server-side only SSR integratation, meaning that every PocketBase API call must be made in a nuxt server-side handler and not in the browser.

If you want to have a mixed approach, aka. to use the SDK in a client-side component and in some node load action, you'll need to manage 2 PocketBase instances similar to the example in https://github.com/pocketbase/js-sdk/issues/29#issuecomment-1229959528, **or at least provide 2 ways to get and set the cookie**. Note that in order to have access to the cookie both from js executed in the browser and js executed in the node server, you'll have to export the cookie with `client.authStore.exportToCookie({ httpOnly: false })`.

When you are inside a node context you have to use `nuxtApp.ssrContext.event.req` and `nuxtApp.ssrContext.event.res` objects to get and set cookies [as shown in the nuxt example](https://github.com/pocketbase/js-sdk#ssr-integration).

When you are inside a browser context, you'll have to get and set the the cookie from `document.cookie`.

If you haven't already, you could also check the `nuxtApp` interface in https://v3.nuxtjs.org/guide/going-further/internals#the-nuxtapp-interface (note that some properties are marked as "server-side only").

**Reply by RA9** [reply]
Thanks, for your quick response. I have been trying to make use of the server-side option but with nuxt 3, plugins aren't loaded on the server side which invalidates the reason for using the plugin method regarding pocketbase.

**Reply by ganigeorgiev** [reply]
@RA9 I don't think this is correct. I don't have much experience with Nuxt 3, but from the docs plugins are allowed for both client and server. I've just found also that you can even prefix the plugin with `.client` and `.server` to restrict the execution context. From https://v3.nuxtjs.org/guide/directory-structure/plugins#creating-plugins:

> Nuxt automatically reads the files in your plugins directory and loads them at the creation of the Vue application. You can use **.server** or **.client** suffix in the file name to load a plugin only on the server or client side.

You can use the original nuxt3 example, but you'll have to add a conditional `loadFromCookie` and `exportToCookie` that will work with both `nuxtApp.ssrContext.event.*` and document.cookie as explained above.

If you want to have only a server-side option, then use the above example and optionally prefix your plugin with `.server` so that the SDK instance is not initialized unnecessary for the client.
